src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/builder/policy/PolicyClearedMessageBuilder.java

```diff
@@ -0,0 +1,156 @@
+                                                                              .applyProviderCreationTime(providerCreationDate);
+            Map<String, PolicyInfo> policyUrlToInfoMap = violationContent.getPolicyInfos().stream().collect(Collectors.toMap(PolicyInfo::getPolicy, Function.identity()));
+            SetMap<ComponentVersionStatus, PolicyInfo> componentPolicies = policyCommonBuilder.createComponentToPolicyMapping(violationContent.getComponentVersionStatuses(), policyUrlToInfoMap);
+            List<ComponentItem> items = new LinkedList<>();
+            for (Map.Entry<ComponentVersionStatus, Set<PolicyInfo>> componentToPolicyEntry : componentPolicies.entrySet()) {
+                ComponentVersionStatus componentVersionStatus = componentToPolicyEntry.getKey();
+                Set<PolicyInfo> policies = componentToPolicyEntry.getValue();
+                final List<ComponentItem> componentItems = retrievePolicyItems(responseCache, componentVersionStatus, policies, notificationId, operation, violationContent.getProjectVersion());
+                items.addAll(componentItems);
+            }
+            projectVersionMessageBuilder.applyAllComponentItems(items);
+            return List.of(projectVersionMessageBuilder.build());
+        } catch (AlertException ex) {
+            logger.error("Error creating policy violation cleared message.", ex);
+        }
+
+        return List.of();
+    }
+
+    private List<ComponentItem> retrievePolicyItems(BlackDuckResponseCache blackDuckResponseCache, ComponentVersionStatus componentVersionStatus,
+        Set<PolicyInfo> policies, Long notificationId, ItemOperation operation, String projectVersionUrl) {
+        List<ComponentItem> componentItems = new LinkedList<>();
+        String componentName = componentVersionStatus.getComponentName();
+        String componentVersionName = componentVersionStatus.getComponentVersionName();
+        ComponentData componentData = new ComponentData(componentName, componentVersionName, projectVersionUrl);
+        componentItems.addAll(policyCommonBuilder.retrievePolicyItems(blackDuckResponseCache, componentData, policies, notificationId, operation,
+            componentVersionStatus.getBomComponent(), List.of()));
+
+        for (PolicyInfo policyInfo : policies) {
+            LinkableItem policyNameItem = componentBuilderUtil.createPolicyNameItem(policyInfo);
+            Optional<PolicyRuleView> optionalPolicyRule = blackDuckResponseCache.getPolicyRule(blackDuckResponseCache, policyInfo);
+            if (optionalPolicyRule.isPresent() && policyCommonBuilder.hasVulnerabilityRule(optionalPolicyRule.get())) {
+                Optional<ComponentItem> vulnerabilityComponent = createEmptyVulnerabilityItem(blackDuckResponseCache, policyNameItem, componentData, notificationId, operation);
```

Comment:
We could avoid the variable assignment on this by just calling `ifPresent` on the method (on a separate line of course).